test/src/engine/BMGameTest.php

```diff
@@ -5759,6 +5773,10 @@ public function test_berserk_round() {
         $this->assertGreaterThanOrEqual(1, $game->actionLog[1]->params['postAttackDice']['attacker'][0]['value']);
 
         $game->update_game_state();
+        $this->assertEquals(BMGameState::CHOOSE_TURBO_SWING, $game->gameState);
```

Comment:
At the moment, every game goes through every game state that exists, and if the game state is irrelevant, the code moves on to the next game state.